### ISSUE TITLE
I want to run the code in a single code step without re-deploying my whole workflow

### ISSUE THREAD

**Issue by dylburger** [PipedreamHQ/roadmap#68]

**Comment by casret** [comment]
Not clear on the ask here, is it more about speeding up the deployment or about running the same pipeline from that step with all the state of a previous event?

**Comment by dylburger** [comment]
Yes I apologize on the lack of detail, this was a placeholder based on a user request and needs more discussion.

The specific ask was for a more interactive editing experience. If I'm playing around with a new npm package whose API I'm still learning, for example, I'd like to run a specific step interactively, as I make changes, without having to re-deploy and re-run the whole workflow.

I realize handling state complicates this (do we use the values from the most recent event, the test event?), but that's the general ask.

**Comment by sauvegardezvous** [comment]
like the idea also,

**Comment by dylburger** [comment]
From @sauvegardezvous in #328 (closed that ticket as a duplicate):

> I am always waiting to test my code in every steps when I am building a workflow.
I have to manipulate data and all previous steps are triggered and do things normally.
And that creates so many unwanted requests to 3rd parties apps.

> Describe the solution you'd like
I suggest to be able to run only one step from all steps, and get the last result available for that step.

> Do you have a workaround?
no, and I can trigger off all others steps but I lost the result (output)
